File: PHP/new_approach/new_approach.php
<?php
	function reverse_number(string $num) : string
	{
		$len = strlen($num);
		$bound = intdiv($len, 2);
		$i = 0;
		$symmetry = $len - 1 - $i;
		$temp = '';

		for ($i = 0; $i < $bound; $i++)
		{
			$symmetry = $len - 1 - $i;
			$temp = $num[$i];
			$num[$i] = $num[$symmetry];
			$num[$symmetry] = $temp;
		}

		return $num;
	}

	$num = readline("Please insert a number: ");
	$res = reverse_number($num);
	echo $res;
?>

File: Ruby/new_approach/new_approach.rb
def reverse_number(num)
	len = num.size
	bound = 0
	bound = len / 2
	symmetry = len - 1
	temp = ''

	bound.times do |i|
		symmetry = len - 1 - i
		temp = num[i]
		num[i] = num[symmetry]
		num[symmetry] = temp
	end

	num
end

print "Please insert a number: "
num = gets.chomp
res = reverse_number num
puts res